src/benchmarks/micro/corefx/System.Text.Json/Utf8JsonReader/Utf8JsonReader.MultiSegment.Tests.cs

```diff
@@ -0,0 +1,283 @@
+
+        [Benchmark(OperationsPerInvoke = Iterations)]
+        public void MultiSegment_SingleShortMultiLineCommentSequence100_CommentHandling_Skip() => MultiSegment_GenericReadTest(s_singleShortMultiLineCommentSequence100, JsonCommentHandling.Skip);
+
+        [Benchmark(OperationsPerInvoke = Iterations)]
+        public void MultiSegment_MultipleShortMultiLineCommentSequence100_CommentHandling_Skip() => MultiSegment_GenericReadTest(s_multipleShortMultiLineCommentSequence100, JsonCommentHandling.Skip);
+
+        [Benchmark(OperationsPerInvoke = Iterations)]
+        public void MultiSegment_SingleLongMultiLineComment1Sequence100_CommentHandling_Skip() => MultiSegment_GenericReadTest(s_singleLongMultiLineComment1Sequence100, JsonCommentHandling.Skip);
+
+        [Benchmark(OperationsPerInvoke = Iterations)]
+        public void MultiSegment_SingleLongMultiLineComment2Sequence100_CommentHandling_Skip() => MultiSegment_GenericReadTest(s_singleLongMultiLineComment2Sequence100, JsonCommentHandling.Skip);
+
+        [Benchmark(OperationsPerInvoke = Iterations)]
+        public void MultiSegment_SingleEmptySingleLineCommentSequence100_CommentHandling_Skip() => MultiSegment_GenericReadTest(s_singleEmptySingleLineCommentSequence100, JsonCommentHandling.Skip);
+
+        [Benchmark(OperationsPerInvoke = Iterations)]
+        public void MultiSegment_MultipleEmptySingleLineCommentSequence100_CommentHandling_Skip() => MultiSegment_GenericReadTest(s_multipleEmptySingleLineCommentSequence100, JsonCommentHandling.Skip);
+
+        [Benchmark(OperationsPerInvoke = Iterations)]
+        public void MultiSegment_SingleShortSingleLineCommentSequence100_CommentHandling_Skip() => MultiSegment_GenericReadTest(s_singleShortSingleLineCommentSequence100, JsonCommentHandling.Skip);
+
+        [Benchmark(OperationsPerInvoke = Iterations)]
+        public void MultiSegment_MultipleShortSingleLineCommentSequence100_CommentHandling_Skip() => MultiSegment_GenericReadTest(s_multipleShortSingleLineCommentSequence100, JsonCommentHandling.Skip);
+
+        [Benchmark(OperationsPerInvoke = Iterations)]
+        public void MultiSegment_SingleLongSingleLineCommentSequence100_CommentHandling_Skip() => MultiSegment_GenericReadTest(s_singleLongSingleLineCommentSequence100, JsonCommentHandling.Skip);
+
+        private static void MultiSegment_GenericReadTest(ReadOnlySequence<byte> data, JsonCommentHandling commentHandling)
+        {
+            for (int i = 0; i < Iterations; i++)
+            {
+                var state = new JsonReaderState(options: new JsonReaderOptions { CommentHandling = commentHandling });
+                var json = new Utf8JsonReader(data, isFinalBlock: true, state);
+
+                while (json.Read())
+                {
+                }
+            }
+        }
+
+        private static ReadOnlySequence<byte> GetSequence(byte[] dataUtf8, int segmentSize)
```

Comment:
GetSequence and dependants (everything to the bottom of this file) are copied over from test helpers in corefx